### PR TITLE
[v6r17] Get list of sites from CS

### PR DESCRIPTION
**Pull request by phicharp**
In preparation for the LHCb change of CERN's name, get list of sites from CS via methods in DMSHelpers().
Note: first commit (replace) is just due to a mistake of mine, no change related... Sorry...

Added a small fix to the logging of the TaskManagerBase (timing was wrong)

In TaskManagerPlugin, allow to add automatically sites with storage using the DMSHelpers() method above. The current implementation doesn't allow to run jobs where data is unless the site is explicitly added in the CS, which is not very good for LHCb use case (Tier2D sites should then be added "by hand")

### CODE REVIEW

File: tests/Workflow/Regression/helloWorld.py [hunk 1]
```diff
@@ -5,21 +5,18 @@
...
 j.setInputSandbox( ['/afs/cern.ch/user/f/fstagni/userJobs/_inputHello.tar.bz2', '/afs/cern.ch/user/f/fstagni/userJobs/hello-script.py'] )
 
-j.setExecutable("exe-script.py","","Ganga_Executable.log")
+j.setExecutable( "exe-script.py", "", "Ganga_Executable.log" )
 
 # <-- user settings
-j.setCPUTime(172800)
-j.setBannedSites(['LCG.CERN.ch', 'LCG.CNAF.it', 'LCG.GRIDKA.de',
-'LCG.IN2P3.fr', 'LCG.NIKHEF.nl', 'LCG.PIC.es', 'LCG.RAL.uk',
-'LCG.SARA.nl'])
```
Comment (fstagni) [inline tests/Workflow/Regression/helloWorld.py]: Why removing this list?

**Comment by chaen** [comment]
Wrong target branch

**Comment by fstagni** [comment]
I changed the target branch to v6r16

**Comment by phicharp** [comment]
PR should be re-reviewed ;-)

File: DataManagementSystem/Utilities/DMSHelpers.py [hunk 2]
```diff
@@ -165,6 +165,29 @@ def getSites( self ):
     self.getSiteSEMapping()
     return sorted( self.siteSet )
 
+  def getTiers( self, withStorage = False, tier = None ):
+    sites = sorted( self.getShortSiteNames( withStorage = withStorage, tier = tier ).values() )
+    if sites and isinstance( sites[0], list ):
```
Comment (chaen) [inline DataManagementSystem/Utilities/DMSHelpers.py]: are you sure that sites will always have at least one element ?

Comment (chaen) [inline DataManagementSystem/Utilities/DMSHelpers.py]: In fact no, if getSiteSEMapping has an error

File: DataManagementSystem/Utilities/DMSHelpers.py [hunk 3]
```diff
@@ -165,6 +165,29 @@ def getSites( self ):
...
+      sites = [s for sl in sites for s in sl]
+    return sites
+
+  def getShortSiteNames( self, withStorage = True, tier = None ):
+    siteDict = {}
+    result = self.getSiteSEMapping()
+    if result['OK']:
+      for site in self.siteSEMapping[LOCAL] if withStorage else self.siteSet:
+        grid, shortSite, _country = site.split( '.' )
+        if isinstance( tier, ( int, long ) ) and ( grid != 'LCG' or gConfig.getValue( '/Resources/Sites/%s/%s/MoUTierLevel' % ( grid, site ), 999 ) != tier ):
```
Comment (chaen) [inline DataManagementSystem/Utilities/DMSHelpers.py]: What is this 999 ? Can we put a constant somewhere ? And a doc ? :)

File: TransformationSystem/Client/TaskManagerPlugin.py [hunk 4]
```diff
@@ -112,9 +113,14 @@ def _ByJobType( self ):
     jobType = self.params['JobType']
     if not jobType:
       raise RuntimeError( "No jobType specified" )
-    excludedSites = self.opsH.getValue( 'JobTypeMapping/%s/Exclude' % jobType, [] )
+    excludedSites = set( self.opsH.getValue( 'JobTypeMapping/%s/Exclude' % jobType, [] ) )
     gLogger.debug( "Explicitly excluded sites for %s task: %s" % ( jobType, ','.join( excludedSites ) ) )
-    excludedSites += self.opsH.getValue( 'JobTypeMapping/AutoAddedSites', [] )
+    autoAddedSites = set( self.opsH.getValue( 'JobTypeMapping/AutoAddedSites', [] ) )
+    if 'WithStorage' in autoAddedSites:
```
Comment (chaen) [inline TransformationSystem/Client/TaskManagerPlugin.py]: what is this ?

File: DataManagementSystem/Utilities/DMSHelpers.py (hunk 2 above)
Comment (sposs) [inline DataManagementSystem/Utilities/DMSHelpers.py]: the ```if sites``` prevents 0-length lists to pass.

Comment (chaen) [inline DataManagementSystem/Utilities/DMSHelpers.py]: Obviously.... sorry, went too fast !

**Comment by fstagni** [comment]
Please merge with v6r17 as it has conflicts now. Also please answer to the comments.

**Comment by phicharp** [comment]
I'll fix conflicts tomorrow...
I would have appreciated to be consulted before moving this PR to v6r17. The changes are trivial and necessary...

**Comment by fstagni** [comment]
Sorry Philippe, it was me suggesting to move to v6r17, for simplicity. They are going in production ASAP.

**Comment by fstagni** [comment]
Can you please reply to the comments? It will be reviewed OK after.

**Comment by phicharp** [comment]
I did reply to all comments, didn't I?

File: TransformationSystem/Client/TaskManagerPlugin.py [hunk 5]
```diff
@@ -123,40 +129,27 @@ def _ByJobType( self ):
...
       gLogger.verbose( res['Message'] )
       allowed = {}
     else:
-      allowed = res['Value']
-      for site in allowed:
-        allowed[site] = fromChar( allowed[site] )
+      allowed = dict( ( site, set( fromChar( fromSites ) ) ) for site, fromSites in res['Value'].iteritems() )
 
     # 5. add autoAddedSites, if requested
-    autoAddedSites = self.opsH.getValue( 'JobTypeMapping/AutoAddedSites', [] )
```
Comment (fstagni) [inline TransformationSystem/Client/TaskManagerPlugin.py]: Removing this line is a bug, it seems to me (`autoAddedSites` won't be defined anymore)

**Pull request merged**